### DISCUSSION TITLE
Error connecting to MSSQL Server when running TestContainers via Gitlab CI

### DISCUSSION THREAD

**Discussion by ryan-buckman** [testcontainers/testcontainers-dotnet#557] [Q&A]
Hi guys,

I'm trying to setup a simple proof of concept running `TestContainers` via `Gitlab CI`, unfortunately I've run into a bit of a problem using the  pre-configured `Microsoft SQL Server` container. I am using the tests found in [MsSqlTestcontainerTests](https://github.com/testcontainers/testcontainers-dotnet/blob/develop/tests/Testcontainers.Tests/Unit/Containers/Unix/Modules/Databases/MsSqlTestcontainerTest.cs) - the tests  pass locally but when run via my `Gitlab CI` pipeline I get the following error

`Sqlcmd: Error: Microsoft ODBC Driver 17 for SQL Server : Login timeout expired.
Sqlcmd: Error: Microsoft ODBC Driver 17 for SQL Server : TCP Provider: Error code 0x2AF9.
Sqlcmd: Error: Microsoft ODBC Driver 17 for SQL Server : A network-related or instance-specific error has occurred while establishing a connection to SQL Server. Server is not found or not accessible. Check if instance name is correct and if SQL Server is configured to allow remote connections. For more information see SQL Server Books Online..
[xUnit.net 00:00:46.22]   IntegrationTests.MsSqlServerTest.ExecScriptInRunningContainer [FAIL]`

Docker variables:
`DOCKER_HOST: tcp://docker:2375`
`DOCKER_TLS_CERTDIR: ""`
`DOCKER_TLS_VERIFY: 0`

Any help would be greatly appreciated.

**Comment by HofmeisterAn** [comment]
Can you add the connection string you get on Gitlab CI?

**Reply by ryan-buckman** [reply]
`msSqlFixture.Container.ConnectionString: Server=docker,49154;Database=master;User Id=sa;Password=yourStrong(!)Password;`
`msSqlFixture.Connection.ConnectionString: Server=docker,49154;Database=master;User Id=sa;`
